kore/src/main/scala/org/kframework/definition/outer.scala

```diff
@@ -372,10 +372,20 @@ case class Module(val name: String, val imports: Set[Import], localSentences: Se
     if (sentences != other)
       return false
 
-    val atts = sentences.collect({ case p: Production => p.att }).toSet
-    val otherAtts = other.collect({ case p: Production => p.att }).toSet
-    val diff = atts -- otherAtts
-    diff.isEmpty
+    val productionsSorted = sentences.collect({ case p: Production => p }).toSeq.sorted
+    val otherProductionSorted = other.collect({ case p: Production => p }).toSeq.sorted
+
+    if (productionsSorted != otherProductionSorted)
+      return false
+
+    for (i <- productionsSorted.indices) {
+      val p1 = productionsSorted(i)
+      val p2 = otherProductionSorted(i)
+      if (!p1.att.equals(p2.att))
```

Comment:
`Ordering` was already defined for `Production`, and in this case, it seems to be to only subclass from `Sentence` that we care about to check and modify the attributes if necessary. 

The comparison between `Attributes` is also already defined in the Attributes object.